### DISCUSSION TITLE
Forecast Notification display issue on old Android versions

### DISCUSSION THREAD

**Discussion by bpezer** [breezy-weather/breezy-weather#1048] [Q&A]
Greetings everybody, I'm writing here because I'm on version 5.2.1_freenet and the screenshots I took aren't in English so I couldn't submit an Issue. In any case it's a banal problem, which I have already fixed on my fork of the application, I would just like to know if I could open a pull request for this (and if not, maybe somebody else could take care of this detail).

While using the application I noticed that the daily forecast notifications were displayed too close to the notifications' top-left edges and it doesn't look particularly pretty. This happens both on my Samsung Galaxy S9 device and on the Android Studio emulator, running the most up-to-date version of the source code. The notifications for the daily forecast currently look as in the below screenshot:

![IMG-20240509-WA0017](https://github.com/breezy-weather/breezy-weather/assets/47884610/78623ff6-1dde-41fb-8410-3cb0534bb1b3)

---

I took a quick glance at the source code and noticed that the relevant XML file was missing some padding. After adding it (16dp) the notification looks like this:

![IMG-20240509-WA0016](https://github.com/breezy-weather/breezy-weather/assets/47884610/277cdad5-dec7-430e-bede-367d049a7fa8)

**Comment by papjul** [comment]
> I'm on version 5.2.1_freenet and the screenshots I took aren't in English so I couldn't submit an Issue

You can easily update from a downloaded APK and switch language in `Settings` > `Appearance`.

Regarding your issue, I don't know what to say, we didn't change anything for a long time and everything is working fine here:
![Screenshot_20240510-073109_Trebuchet_1](https://github.com/breezy-weather/breezy-weather/assets/1449159/3054f57a-5fe3-4479-b19f-3544a131e86c)

Something doesn't feel very native to me, is it a custom ROM or something?

**Reply by min7-i** [reply]
That's also happening on some stock Android versions below A12. My assumption is that this is caused by the changes made [here](https://github.com/breezy-weather/breezy-weather/commit/0f053b6b24735390b9267f3c0d96f523ccbfff6a).

However, instead of modifying the [XML layout](https://github.com/breezy-weather/breezy-weather/blob/main/app/src/main/res/layout/notification_forecast.xml), I'd rather replace the [custom big content view](https://github.com/breezy-weather/breezy-weather/blob/08738a7bdd2496123d7a804c52e1e0aca1275681/app/src/main/java/org/breezyweather/background/forecast/ForecastNotificationNotifier.kt#L105) with [BigTextStyle](https://developer.android.com/reference/androidx/core/app/NotificationCompat.BigTextStyle) for setting the content of the expanded notification. That way the notification should be displayed properly in all versions.

I could take a closer look at this at the weekend or do you want to do this, @RiffLord?

**Comment by bpezer** [comment]
> That's also happening on some stock Android versions below A12. My assumption is that this is caused by the changes made [here](https://github.com/breezy-weather/breezy-weather/commit/0f053b6b24735390b9267f3c0d96f523ccbfff6a).

My phone runs stock Android 10, so I guess that's why I've got this problem. 

> I could take a closer look at this at the weekend or do you want to do this, @RiffLord?

If it's alright I'd like to give it a try, I'm looking to get back into coding after a long break.

**Reply by min7-i** [reply]
@RiffLord Are you still interested in working on this? I don't want to pressure you. So if you want to do this, feel free to take your time.

**Reply by bpezer** [reply]
Hey there, I'm very sorry for disappearing but the past few weeks have been much more hectic than I anticipated. I'd still like to give it a go if it's not too pressing.

**Reply by min7-i** [reply]
Thanks for the quick update. In my opinion it's not that urgent. It's only affecting older Android versions and it's not like forecast notifications are unusable.

> I'm very sorry for disappearing but the past few weeks have been much more hectic than I anticipated.

We're all working on the project in our free time. So take the time you need.

**Comment by bpezer** [comment]
> You can easily update from a downloaded APK and switch language in `Settings` > `Appearance`.

Would this be the recommended way to keep the application up to date? I got it from F-Droid and have seen occasional updates since I've started using it, so I assumed it was the "official" version.

**Reply by papjul** [reply]
All distribution channels are here: https://github.com/breezy-weather/breezy-weather/blob/main/INSTALL.md

The file distributed on F-Droid default report is exactly the same as the one on GitHub, you can check the checksum, except F-Droid default repo is much slower to get updates. Once their pipeline confirmed the reproducibility and various checks, you will unfortunately need to wait for the current build cycle to end, and for the next one to start. Each build cycle takes about 4 days when everything is working great (which is not always the case), so the usually optimist case is 1 week starting from the moment the update is added to the metadata. I really don't like that, but it was a popular request, so it was added there. I'm only asking in return that people updates their version before they report bugs so I don't spend time investigating something that was fixed in a newer version. It takes less than a minute to do, doesn’t need uninstall, just install the update on top, and done.

**Reply by bpezer** [reply]
Understood. I'll apply the update and see if there's any difference.

**Reply by papjul** [reply]
In your case, there is no difference, especially given you said yourself that you compiled from source, which is even newer ;)

**Reply by bpezer** [reply]
This is of course very true :D
Then I guess it's due to me running an older version of Android, as mentioned above. I'll see if the suggested fix works over the weekend.

**Comment by min7-i** [comment]
This was resolved by #1322. So I guess it can be closed.
